collector/consumer/processor/aggregateprocessor/internal/defaultaggregator/value_recorder.go

```diff
@@ -29,41 +26,32 @@ func (r *valueRecorder) Record(key *internal.LabelKeys, gaugeValues []*model.Gau
 	if key == nil {
 		return
 	}
-	r.mutex.RLock()
-	aggValues, ok := r.labelValues[*key]
-	r.mutex.RUnlock()
+	aggValues, ok := r.labelValues.Load(*key)
 	if !ok {
-		r.mutex.Lock()
 		// double check to avoid double writing
-		aggValues, ok = r.labelValues[*key]
-		if !ok {
-			aggValues = newAggValuesMap(gaugeValues, r.aggKindMap)
-			r.labelValues[*key] = aggValues
-		}
-		r.mutex.Unlock()
+		aggValues, _ = r.labelValues.LoadOrStore(*key, newAggValuesMap(gaugeValues, r.aggKindMap))
 	}
```

Comment:
No double check anymore

Reply:
`LoadOrStore` is actually the "double-check", otherwise `Store` will be used.